src/Features/Core/Portable/Workspace/CompileTimeSolutionProvider.cs

```diff
@@ -44,20 +45,18 @@ public Factory()
         private const string RazorSourceGeneratorTypeName = "Microsoft.NET.Sdk.Razor.SourceGenerators.RazorSourceGenerator";
         private static readonly string s_razorSourceGeneratorFileNamePrefix = Path.Combine(RazorSourceGeneratorAssemblyName, RazorSourceGeneratorTypeName);
 
-        private readonly Workspace _workspace;
-
         private readonly object _gate = new();
 
         /// <summary>
-        /// Cached compile time solution corresponding to the <see cref="Workspace.PrimaryBranchId"/>
+        /// Cached compile-time solution corresponding to an existing design-time solution.
         /// </summary>
-        private (int DesignTimeSolutionVersion, BranchId DesignTimeSolutionBranch, Solution CompileTimeSolution)? _primaryBranchCompileTimeCache;
+#if NETCOREAPP
+        private readonly ConditionalWeakTable<Solution, Solution> _designTimeToCompileTimeSolution = new();
+#else
+        private ConditionalWeakTable<Solution, Solution> _designTimeToCompileTimeSolution = new();
+#endif
```

Comment:
.net core has `ConditionalWeakTable.Clear()`.  Standard does not.  So we simulate it by overwriting the field.